custom_components/mammotion/switch.py

```diff
@@ -116,23 +121,21 @@ def add_entities() -> None:
         new_areas = (set(areas) | set(area_name_hashes)) - added_areas
         if new_areas:
             for area_id in new_areas:
-                existing_name: AreaHashNameList = next(
+                existing_name: AreaHashNameList | None = next(
                     (area for area in area_name if str(area.hash) == str(area_id)), None
                 )
-                name = (
-                    existing_name.name
-                    if (existing_name is None or existing_name != "")
-                    else f"Area {area_id}"
-                )
+                name = existing_name.name if existing_name else f"{area_id}"
```

Comment:
I may have messed the logic up here a little, existing_name.name could be "" so it also needs to check if existing_name.name
```suggestion
                name = existing_name.name if (existing_name and existing_name.name) else f"{area_id}"
```